services/server/src/server/services/storageServices/S3RepositoryService.ts

```diff
@@ -0,0 +1,93 @@
+import { RepositoryV2Service } from "./RepositoryV2Service";
+import { WStorageService } from "../StorageService";
+import { WStorageIdentifiers } from "./identifiers";
+import logger from "../../../common/logger";
+import {
+  S3Client,
+  PutObjectCommand,
+  ListObjectsV2Command,
+  DeleteObjectsCommand,
+} from "@aws-sdk/client-s3";
+import { PathConfig } from "../../types";
+import Path from "path";
+
+export interface S3RepositoryServiceOptions {
+  s3Bucket: string;
+  s3Region: string;
+  s3AccessKeyId: string;
+  s3SecretAccessKey: string;
+  endpoint?: string;
+}
+
+export class S3RepositoryService
+  extends RepositoryV2Service
+  implements WStorageService
+{
+  IDENTIFIER = WStorageIdentifiers.S3Repository;
+  private s3: S3Client;
+  private bucket: string;
+
+  constructor(options: S3RepositoryServiceOptions) {
+    super({ ipfsApi: "", repositoryPath: "" });
```

Comment:
`ipfsApi` is not used anymore. Would be nice to remove it also from the `RepositoryV2Service`.